client/landing/stepper/declarative-flow/copy-site.tsx

```diff
@@ -0,0 +1,139 @@
+import { ONBOARD_STORE } from 'calypso/landing/stepper/stores';
+import { recordFullStoryEvent } from 'calypso/lib/analytics/fullstory';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import {
+	clearSignupDestinationCookie,
+	setSignupCompleteSlug,
+	persistSignupDestination,
+	setSignupCompleteFlowName,
+} from 'calypso/signup/storageUtils';
+import { useSiteSlug } from '../hooks/use-site-slug';
+import { recordSubmitStep } from './internals/analytics/record-submit-step';
+import AutomatedCopySite from './internals/steps-repository/automated-copy-site';
+import Intro from './internals/steps-repository/intro';
+import ProcessingStep, { ProcessingResult } from './internals/steps-repository/processing-step';
+import SiteCreationStep from './internals/steps-repository/site-creation-step';
+import type { Flow, ProvidedDependencies } from './internals/types';
+
+const copySite: Flow = {
+	name: COPY_SITE_FLOW,
+
+	get title() {
+		return translate( 'Copy Site' );
+	},
+
+	useSteps() {
+		useEffect( () => {
+			recordTracksEvent( 'calypso_signup_start', { flow: this.name } );
+			recordFullStoryEvent( 'calypso_signup_start_copy_site', { flow: this.name } );
+		}, [] );
+
+		return [
+			{ slug: 'intro', component: Intro },
+			{ slug: 'siteCreationStep', component: SiteCreationStep },
+			{ slug: 'processing', component: ProcessingStep },
+			{ slug: 'automatedCopy', component: AutomatedCopySite },
+		];
+	},
+
+	useStepNavigation( _currentStepSlug, navigate ) {
+		const flowName = this.name;
+		const { setStepProgress } = useDispatch( ONBOARD_STORE );
+		const flowProgress = useFlowProgress( { stepName: _currentStepSlug, flowName } );
+		const siteSlug = useSiteSlug();
+		const urlQueryParams = useQuery();
+
+		const { setPlanCartItem } = useDispatch( ONBOARD_STORE );
+
+		setStepProgress( flowProgress );
+
+		const submit = async (
+			providedDependencies: ProvidedDependencies = {},
+			...params: string[]
+		) => {
+			recordSubmitStep( providedDependencies, '', flowName, _currentStepSlug );
+
+			switch ( _currentStepSlug ) {
+				case 'intro': {
+					clearSignupDestinationCookie();
+					return navigate( 'siteCreationStep' );
+				}
+
+				case 'siteCreationStep': {
+					return navigate( 'processing' );
+				}
+
+				case 'processing': {
+					if ( providedDependencies?.finishedWaitingForAtomic ) {
+						return window.location.assign( `/home/${ siteSlug }` );
+					}
+					const processingResult = params[ 0 ] as ProcessingResult;
+
+					if ( processingResult === ProcessingResult.FAILURE ) {
+						return navigate( 'error' );
+					}
+
+					const productSlug = urlQueryParams.get( 'productSlug' );
+					if ( ! productSlug ) {
+						throw new Error( 'Plan product not found' );
+					}
+					const productToAdd = {
+						product_slug: productSlug,
+					};
+
+					setPlanCartItem( productToAdd );
+
+					const destination = addQueryArgs( `/setup/${ this.name }/automatedCopy`, {
```

Comment:
Are `camelCase` URLs common elsewhere? Or can we switch to hyphens? `automated-copy`? 

Also, for this particular case, we could switch to `copy-site`?

Reply:
Done here: https://github.com/Automattic/wp-calypso/commit/120a74f3c1f555c207af7f8a77fdddbb4785e827
Almost all step URLs in other flows use camel case. However, I also prefer URLs with dashes, so I've changed them to use snake case.
The name is `automated-copy` because `copy-site` is used to identify the flow. The URL currently is similar to:
`/setup/copy-site/automated-copy`